recipe/run_test.py

```diff
@@ -1,16 +1,39 @@
 import os
 import sys
 import ctypes
+import sqlite3
 
 if sys.platform == 'win32':
     path = os.path.join(sys.prefix, 'Library', 'bin', 'spatialite.dll')
-    libfreexl = ctypes.CDLL(path)
 elif sys.platform == 'darwin':
     # LD_LIBRARY_PATH not set on OS X or Linux.
     path = os.path.expandvars('$PREFIX/lib/libspatialite.dylib')
-    libfreexl = ctypes.CDLL(path)
 elif sys.platform.startswith('linux'):
     path = os.path.expandvars('$PREFIX/lib/libspatialite.so')
-    libfreexl = ctypes.CDLL(path)
 else:
     raise Exception('Cannot recognize platform {!r}'.format(sys.platform))
+# make sure the dynamic library can be found and loads
+libfreexl = ctypes.CDLL(path)
+
+if sys.platform == 'win32':
+    path = os.path.join(sys.prefix, 'Library', 'bin', 'mod_spatialite.dll')
+elif sys.platform == 'darwin':
+    # LD_LIBRARY_PATH not set on OS X or Linux.
+    path = os.path.expandvars('$PREFIX/lib/mod_spatialite.dylib')
+elif sys.platform.startswith('linux'):
+    path = os.path.expandvars('$PREFIX/lib/mod_spatialite.so')
+else:
+    raise Exception('Cannot recognize platform {!r}'.format(sys.platform))
+# make sure the dynamic library can be found and loads
+libfreexl = ctypes.CDLL(path)
+
+# Test loading the resulting SQLite3 extension
+db = sqlite3.connect(':memory:')
+db.enable_load_extension(True)
+if sys.platform == 'win32':
+    # not loading successfully without this
+    db.execute("SELECT load_extension(?, 'spatialite_init_ex')", [path])
+else:
+    db.execute("SELECT load_extension(?)", [path])
```

Comment:
I'm not super happy about this but can we live with it for now (if it works)?

Reply:
See https://www.sqlite.org/lang_corefunc.html#load_extension . any idea what "the default entry point name" is? Maybe it isn't `spatialite_init_ex`? Some other function that isn't exported on Windows?

Reply:
Looking here: https://www.sqlite.org/c3ref/load_extension.html
the default seems to be: `sqlite3_extension_init`.  Presumably, then, the problem is that Windows doesn't have this function?

Reply:
Okay, so it looks like the fallback name is `sqlite3_spatialite_init`.  This should be getting declared in `spatialite.h` and this would seem to be happening in Linux and OSX but nto in Windows.  So I think we still have a problem.  I don't think `spatialite_init_ex` is the correct entry point.

The code looks like:
```
#ifdef LOADABLE_EXTENSION
    SPATIALITE_DECLARE int
	sqlite3_spatialite_init (sqlite3 * db, char **pzErrMsg,
				 const sqlite3_api_routines * pApi);
#endif
```
In the `nmake_mod.opt` file, it looks like `LOADABLE_EXTENSION` should indeed be getting set:
```
# Uncomment the first for an optimized build, or the second for debug.
OPTFLAGS=	/nologo /Ox /fp:precise /W4 /MD /D_CRT_SECURE_NO_WARNINGS \
		/DDLL_EXPORT /DLOADABLE_EXTENSION /DYY_NO_UNISTD_H
#OPTFLAGS=	/nologo /Zi /MD /Fdmod_spatialite.pdb /DDLL_EXPORT
```
So I'm still at a bit of a loss.

Reply:
So this isn't in the list of exports from `dumpbin`... 
`SPATIALITE_DECLARE` should be set to `__declspec(dllexport)` due to `DLL_EXPORT` being defined

But, I can't find where `sqlite3_spatialite_init` is implemented. Is declared in `spatialite.h`, but I can't see an implementation. Am I missing something?

Reply:
@PMeira, you mentioned in this issue: https://github.com/ContinuumIO/anaconda-issues/issues/10926 
that you thought `mod_spatialite.dll` was missing various symbols.  Would that include `sqlite3_spatialite_init` and do you have any clue how to get this fixed?

Reply:
> But, I can't find where sqlite3_spatialite_init is implemented. Is declared in spatialite.h, but I can't see an implementation. Am I missing something?

Nor could I.  I was also really confused by that.

Reply:
I looked into this a bit further.  I don't think `sqlite3_spatialite_init` is what is being used.  Instead, it's `sqlite3_modspatialite_init`, which probably makes sense, since the name of the dynamic library is `mod_spatialite`, not `spatialite`.

```
#ifdef LOADABLE_EXTENSION	/* loadable-extension only */
#if !(defined _WIN32) || defined(__MINGW32__)
/* MSVC is unable to understand this declaration */
__attribute__((visibility ("default")))
#endif
     SPATIALITE_DECLARE int
	 sqlite3_modspatialite_init (sqlite3 * db, char **pzErrMsg,
				     const sqlite3_api_routines * pApi)
{
/* SQLite invokes this routine once when it dynamically loads the extension. */
    spatialite_initialize ();
    return init_spatialite_extension (db, pzErrMsg, pApi);
}
#endif
```

Note that this is explicitly being skipped because MSVC doesn't understand it.

I'm not what we can do about this.  I don't know the first thing about `sqlite3`, so I'm just learning a bit as I go.  Is there a way to manually call `spatialite_initialize` and then `init_spatialite_extension`?

Reply:
No, it looks like `init_spatialite_extension` isn't exported so that's out.

Reply:
> Note that this is explicitly being skipped because MSVC doesn't understand it.

I think it's just the `__attribute__((visibility ("default")))` that's being skipped - do you agree?

Reply:
Oh, I see.  Hmm, then what is going wrong?